docs: update FAQ entry about vite dependency pre-bundling (#158)
sveltejs/vite-plugin-svelte: docs/faq.md

For reference, check out [windicss](https://github.com/windicss/vite-plugin-windicss/blob/517eca0cebc879d931c6578a08accadfb112157c/packages/vite-plugin-windicss/src/index.ts#L167).

> ### What is going on with vite and `Pre-bundling dependencies:`?

Reviewer: The docs currently capitalize special names, like Vite, CJS, ES, API. I think we should do that here too. (and some other spots below)

Pre-bundling dependencies is an [optimization in vite](https://vitejs.dev/guide/dep-pre-bundling.html).

> It is required for cjs dependencies, as vite's development server only works with es modules internally.

Reviewer 2: ```suggestion
It is required for CJS dependencies, as Vite's development server only works with ES modules on the client side.
```

> Thanks to [new api in vite](https://github.com/vitejs/vite/pull/4634) , [vite-plugin-svelte](https://github.com/sveltejs/vite-plugin-svelte/pull/157) automatically handles pre-bundling these for you.

Reviewer: ```suggestion
Thanks to [a new API in Vite](https://github.com/vitejs/vite/pull/4634), [vite-plugin-svelte](https://github.com/sveltejs/vite-plugin-svelte/pull/157) automatically handles pre-bundling these for you.
```

> In case you still run into errors like `The requested module 'xxx' does not provide an export named 'yyy'` please check our [open issues](https://github.com/sveltejs/vite-plugin-svelte/issues).

Reviewer: ```suggestion
In case you still run into errors like `The requested module 'xxx' does not provide an export named 'yyy'`, please check our [open issues](https://github.com/sveltejs/vite-plugin-svelte/issues).
```
